runtime/lua/vim/lsp/util.lua

```diff
@@ -1105,7 +1111,31 @@ function M.make_floating_popup_options(width, height, opts)
     or vim.fn.winline() - 1
   local lines_below = vim.fn.winheight(0) - lines_above
 
-  if lines_above < lines_below then
+  local anchor_bias = opts.anchor_bias or 'auto'
+
+  local anchor_below
+
+  if anchor_bias == 'below' then
+    if lines_below > lines_above then
+      anchor_below = true
+    else
+      anchor_below = height <= lines_below
+    end
```

Comment:
ternary can be very concise without harming readability (often improves it).
similar for other cases below.

```suggestion
    anchor_below = (lines_below > lines_above) and true or (height <= lines_below)
```